tests/zfs-tests/tests/functional/channel_program/synctask_core/tst.inherit.ksh

```diff
@@ -0,0 +1,35 @@
+#!/bin/ksh -p
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2019 Joyent, Inc.
+#
+
+. $STF_SUITE/tests/functional/channel_program/channel_common.kshlib
+
+verify_runnable "global"
+
+fs=$TESTPOOL/$TESTFS
+testprop="com.joyent:testprop"
+testval="testval"
+
+log_must zfs set $testprop=$testval $fs
+log_must_program_sync $TESTPOOL - $fs $testprop <<-EOF
+	arg = ...
+	fs = arg["argv"][1]
+	prop = arg["argv"][2]
+	err = zfs.sync.inherit(fs, prop)
+	msg = "resetting " .. prop .. " on " .. fs .. " err=" .. err
+	return msg
+EOF
+
+log_pass "Inherit/clear property with channel program works."
```

Comment:
This test case appears to fail for me, but interestingly doesn't get reported by ZTS as a failure.  You can see this in the CI logs as well, for example http://build.zfsonlinux.org/builders/Amazon%202%20x86_64%20Release%20%28TEST%29/builds/8075/steps/shell_9/logs/log

Adding an additional check which runs after the zcp to verify the property was cleared would be good.  The existing  `synctask_core/tst.destroy_snap.ksh` test case does this.

```
Test: /usr/share/zfs/zfs-tests/tests/functional/channel_program/synctask_core/tst.inherit (run as root) [00:00] [PASS]
18:05:55.78 SUCCESS: zfs set com.joyent:testprop=testval testpool/testfs
18:05:55.78 NOTE: running: zfs program -n  testpool /tmp/tmp.7MuKPZsPsZ testpool/testfs com.joyent:testprop:
18:05:55.79 NOTE: input:
18:05:55.79 arg = ...
18:05:55.79 fs = arg["argv"][1]
18:05:55.79 prop = arg["argv"][2]
18:05:55.79 err = zfs.sync.inherit(fs, prop)
18:05:55.79 msg = "resetting " .. prop .. " on " .. fs .. " err=" .. err
18:05:55.79 return msg
18:05:55.79 NOTE: output:
18:05:55.79 
18:05:55.79 NOTE: error:
18:05:55.79 Channel program execution failed:
18:05:55.79 [string "channel program"]:4: running functions from the zfs.sync submodule requires passing sync=TRUE to lzc_channel_program() (i.e. do not specify the "-n" command line argument)
18:05:55.79 stack traceback:
18:05:55.79 	[C]: in function 'inherit'
18:05:55.79 	[string "channel program"]:4: in main chunk
18:05:55.79 NOTE: running: zfs program  testpool /tmp/tmp.7MuKPZsPsZ testpool/testfs com.joyent:testprop:
18:05:55.80 NOTE: input:
18:05:55.80 arg = ...
18:05:55.80 fs = arg["argv"][1]
18:05:55.80 prop = arg["argv"][2]
18:05:55.80 err = zfs.sync.inherit(fs, prop)
18:05:55.80 msg = "resetting " .. prop .. " on " .. fs .. " err=" .. err
18:05:55.80 return msg
18:05:55.80 NOTE: output:
18:05:55.80 Channel program fully executed and produced output:
18:05:55.80     return: 'resetting com.joyent:testprop on testpool/testfs err=0'
18:05:55.81 NOTE: error:
18:05:55.81 
18:05:55.81 Inherit/clear property with channel program works.
```